File: javascript/23.js
const limit =   28123;
let abundantArray = [];
let abundantSumArray = new Array(limit).fill(0);
let primes=[]

function calcDivisors(n){    
    for(let i=2;i<=n;i++){
        let divisors=[]
        let num = i
        for(let j=0;j<primes.length;j++){
            if(primes[j]>num) break
            let divCounter=0
            while(num%primes[j]==0){
                divCounter++;
                num=num/primes[j]
            }
            if(divCounter>0){
                divisors.push([primes[j],divCounter])
            }            
        }        
        if(num==i) primes.push(i)
        let sum=1
        if(divisors.length>0){            
            divisors.forEach(element => {
                sum*=( (Math.pow(element[0],element[1]+1) -1) /  ( element[0]-1 ) )
            });            
        }                        
        if(sum-i>i) abundantArray.push(i)
    }    
}

calcDivisors(limit)


let result = 0;
for(let a=0;a<abundantArray.length;a++){
    for(let b=a;b+a<abundantArray.length && abundantArray[a]+abundantArray[b] <= abundantSumArray.length;b++){
        abundantSumArray[abundantArray[a]+abundantArray[b]]=1
    }
}

let ans=0
for(let i=0;i<abundantSumArray.length;i++){
    if(abundantSumArray[i]==0)ans+=i
}


console.log(ans);

File: python/23.py
import  math
number = 28123
abundantNums=[]
abunSum=[0]*(number)
primes=[]


def calcDivisors(n):
    i=2
    while i<=n:
        divisors=[]
        num=i

        for j in primes:
            if j>num: break
            divCounter=0
            while num%j==0:
                divCounter+=1
                num/=j
            if divCounter>0:   divisors.append([j, divCounter])
        if num==i:  primes.append(num)

        sum=1
        if len(divisors)>0:
            for x in divisors:
                sum*=( ( (math.pow(x[0],x[1]+1)) -1 ) / ( x[0] -1 ))
        if int(sum)-i > i: abundantNums.append(i)
        i+=1

calcDivisors(number)

for i in range(0,len(abundantNums)):
    j=i
    while abundantNums[i]+abundantNums[j] < number and i+j<len(abundantNums):
        abunSum[abundantNums[i]+abundantNums[j]]=1
        j+=1

ans=0
for i in range(0,len(abunSum)):
    if abunSum[i]==0:  ans+=(i)
print(ans)